modules/javafx.graphics/src/main/java/com/sun/javafx/text/TextRun.java

```diff
@@ -285,15 +285,10 @@ public int getWrapIndex(float width) {
              * should only be used when the run has contextual shaping.
              */
             /* Not need to check for compact as bidi disables the simple case */
-            int gi = 0;
-            float runWidth = positions[glyphCount<<1];
-            while (runWidth > width) {
-                float glyphWidth = positions[(gi+1)<<1] - positions[gi<<1];
-                if (runWidth - glyphWidth <= width) {
+            for (int gi = glyphCount; gi <= 0; gi--) {
```

Comment:
should we also correct the comment on line 288?

Reply:
oops. I think I see the problem.
And I had noticed the typo in the existing comment and can correct that ..

Reply:
Updated to look at the right segment !